src/features/posts/components/add-post-form.tsx

```diff
@@ -1,38 +1,62 @@
 import { useState } from 'react';
+import { useAuth } from '../../../shared/hooks/useAuth';
 import { PostContent } from '../types/post-content';
 import { postContentSchema } from '../utils/validate';
 import { fromZodError } from 'zod-validation-error';
-
+import { Editor } from '@tinymce/tinymce-react';
 import Box from '@mui/material/Box';
 import TextField from '@mui/material/TextField';
 import { Button } from '@mui/material';
+import { ApiKeyTinyMMC } from '../../../shared/config/confidential-data';
+import { useCreatePost } from '../api/use-create-post';
+import { generateID } from '../utils/generate-id';
 
-import ReactQuill from 'react-quill';
-import 'react-quill/dist/quill.snow.css';
-
+import { ref, uploadBytes } from "firebase/storage";
+import { imageDatabase } from '../../../shared/config/firebase-image';
 
-let DEFAULT_POST: PostContent = {
+const DEFAULT_POST: PostContent = {
+    postId: 0,
+    userId: 0,
     title: '',
     content: ''
 }
 
 export const AddPostForm = () => {
     const [content, setContent] = useState<string>('');
     const [error, setError] = useState<PostContent>(DEFAULT_POST);
-    const handleContentChange = (content: string) => setContent(content);
+    const [image, setImage] = useState<any>(null);
+    const { userData } = useAuth();
+    const { mutate } = useCreatePost()
+
+    const handleContentChange = (e: React.FormEvent<HTMLFormElement> | any) => setContent(e.target.getContent());
+    const randomID = generateID(1000000000);
 
+    const handleImageChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        if (e.target.files) {
+            setImage(e.target.files[0])
+        }
+    }  
+    
+    const uploadImage = (image: any) => {
+        uploadBytes(ref(imageDatabase, `posts/${randomID}`), image).then(() => {
+            window.location.reload();
+        })
+    }
 
     const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
         e.preventDefault();
         let postContent: PostContent = {
+            postId: randomID,
+            userId: userData.user_id,
             title: e.currentTarget['post-title'].value,
             content: content
         }
         setError(DEFAULT_POST);
         try{
-            let PostContent = postContentSchema.parse(postContent);
+            let PostContentSchema = postContentSchema.parse(postContent);
```

Comment:
1. const
2. to nie jest PostContentSchema. to jest postContent. PostContentSchema to jest schema do validowania i używasz tego po prawej stronie =, po lewej dostajesz wynik walidacji czyli postContent